sdk/core/core/inc/az_json_read.h

```diff
@@ -61,7 +59,7 @@ typedef struct {
   az_json_stack stack;
 } az_json_state;
 
-AZ_NODISCARD az_json_state az_json_state_create(az_const_span const buffer);
+AZ_NODISCARD az_json_state az_json_state_create(az_span const buffer);
```

Comment:
these may want to be _init methods returning an az_result. Although I suppose if we ever need that the _init name is there.

Reply:
Should we introduce some relaxed rules for internal API to allow `_create` method? It's a little bitannoying to write simple initializers with `az_result` conventions.

Reply:
https://github.com/Azure/azure-sdk-for-c/issues/133